pkg/codegen/go/gen.go

```diff
@@ -4512,15 +4514,33 @@ func packageName(pkg *schema.Package) string {
 	return goPackage(root)
 }
 
-func GeneratePackage(tool string, pkg *schema.Package) (map[string][]byte, error) {
+func GeneratePackage(tool string,
+	pkg *schema.Package,
+	localDependencies map[string]string,
+) (map[string][]byte, error) {
 	if err := pkg.ImportLanguages(map[string]schema.Language{"go": Importer}); err != nil {
 		return nil, err
 	}
 
 	var goPkgInfo GoPackageInfo
 	if goInfo, ok := pkg.Language["go"].(GoPackageInfo); ok {
 		goPkgInfo = goInfo
+	} else {
+		goPkgInfo = GoPackageInfo{}
 	}
+
+	if goPkgInfo.ImportBasePath == "" {
+		if !pkg.SupportPack {
+			goPkgInfo.ImportBasePath = extractImportBasePath(pkg.Reference())
+		} else {
+			var vPath string
```

Comment:
Should this be added to extractImportBasePath? Just ensure "meta" is in the PackageReference interface.

Reply:
Added `SupportPack() : bool` to the package reference and used in `extractImportBasePath`. I avoided full `Meta` object since the `moduleFormat` field is private in `schema.Package`.